src/common/components/repositories-home/index.js

```diff
@@ -19,16 +20,19 @@ const SNAP_POLL_PERIOD = (15 * 1000);
 
 class RepositoriesHome extends Component {
   fetchData(props) {
-    const { snaps } = props;
+    const { hasSnaps, snaps, entities } = props;
 
-    if (snaps.success) {
+    // check both success and fetching to avoid redirecting
+    // based on previous success
+    if (snaps.success && !snaps.isFetching) {
```

Comment:
We had issues with this before and still in some places we have different corner cases that require confusing checks on `success` and/or `isFetching`.

Probably not worth handling in this PR, but we should add it as discussion point.